python/tank/util/local_file_storage.py

```diff
@@ -84,7 +84,9 @@ def get_global_root(cls, path_type, generation=CORE_V18):
                            which is the current generation of paths.
         :return: Path as string
         """
-        if generation == cls.CORE_V18:
+        # If SHOTGUN_HOME is set then the intent is to not use any of official locations and instead
+        # use an alternate location, in which case this is as if we're running core 0.18.
+        if generation == cls.CORE_V18 or os.environ.get("SHOTGUN_HOME"):
```

Comment:
but if the pre-18 behaviour is requested, will the consumer code be able to understand `SHOTGUN_HOME`? I think that's the reason it was done the way it was in the first place. Not a bit deal and not even sure if i understand what that use case would be :) Just pointing it out - i think you wrote the original SHOTGUN_HOME code so you probably know the original reason why it was done the way it was.